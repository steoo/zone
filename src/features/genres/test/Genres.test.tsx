import React from 'react';
import { genres } from './Genres.fixtures';
import { getGenreById } from '../Genres.utils';
import renderWithStore from '../../../app/utils/testing';
import Genre from '../Genre.component';

describe('Features', () => {
  describe('Genres', () => {
    describe('Component', () => {
      it('should render the correct genre', () => {
        const { baseElement, getByText } = renderWithStore(<Genre genreId={28} />, {
          initialState: {
            genres: {
              isFetching: false,
              data: genres
            }
          }
        });

        expect(getByText('Action')).toBeInTheDocument();
        expect(baseElement).toMatchSnapshot();
      });
    });

    describe('Utils', () => {
      it('should return the id by genre', () => {
        expect(getGenreById(genres, 12)).toEqual({
          id: 12,
          name: 'Adventure'
        });
      });
    });
  });
});
