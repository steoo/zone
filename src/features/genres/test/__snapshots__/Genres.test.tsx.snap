// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features Genres Component should render the correct genre 1`] = `
<body>
  <div>
    <button
      class="sc-gsTCUz dOAjnQ"
    >
      Action
    </button>
  </div>
</body>
`;
